docs/okeps/okep-5535-support-multi-nic-and-multi-vtep-in-ic-mode.md

```diff
@@ -0,0 +1,387 @@
+        options: {csum="true"}
+    Encap geneve
+        ip: "10.0.0.2"
+        options: {csum="true"}
+    Encap geneve
+        ip: "10.0.0.3"
+        options: {csum="true"}
+    Port_Binding default.ovn.blue_test_pod-a  # The Port_Binding of Pod-a's `net-blue` network interface
+...
+
+# ovn-sbctl list Port_Binding default.ovn.blue_test_pod-a
+_uuid               : 04fd102a-6679-4920-ad6c-663c52df4161
+chassis             : a02a2083-6bd9-47f6-ab34-730322756c9e
+datapath            : 6413cf4a-838f-481f-b4b9-1ac9ab467cb9
+encap               : 4a64dec0-b51c-4e7b-bf0f-629cce23bf42 # set by ovn-controller according to OVS interface external_ids:encap-ip
+logical_port        : default.ovn.blue_test_pod-a
+tunnel_key          : 11
+...
+
+# ovn-sbctl  list encap 4a64dec0-b51c-4e7b-bf0f-629cce23bf42
+_uuid               : 4a64dec0-b51c-4e7b-bf0f-629cce23bf42
+chassis_name        : node-a
+ip                  : "10.0.0.3"
+options             : {csum="true"}
+type                : geneve
+...
+```
+
+
+
+#### Layer 3 topology solution
+
+In OVN IC mode, Layer 3 networks use Transit Switches to route traffic between nodes. Each remote node is
+represented by a remote logical switch port (LSP) in the network’s Transit Switch.
+
+For example, on node-b, the following remote LSPs represent node-a across three different networks:
+```
+switch 8d303553-017f-45e4-8fb4-1fe549875eb1 (transit_switch)
+   port tstor-node-a
+        type: remote
+        addresses: ["0a:58:64:58:00:06 100.88.0.6/16"]
+
+switch ef9a91a3-85ec-476f-972c-de4de105ff3c (net.red_transit_switch)
+   port net.red_tstor-node-a
+        type: remote
+        addresses: ["0a:58:64:58:00:06 100.88.0.6/16"]
+
+switch a50d8fc0-b0fd-49b8-b5d0-cd718860bf9d (net.blue_transit_switch)
+   port net.blue_tstor-node-a
+        type: remote
+        addresses: ["0a:58:64:58:00:06 100.88.0.6/16"]
+```
+
+Each Transit Switch includes only one remote LSP for node-a, even though node-a has multiple VTEP interfaces.
+Additionally, these LSPs do not have the `encap` field explicitly populated in their corresponding
+`Port_Binding` entries.
+
+##### Supporting Multiple Transit Ports Per Encap IP
+
+In the current implementation, a remote Transit Switch port representing a node uses:
+- `k8s.ovn.org/node-id` as the tunnel key
+- `k8s.ovn.org/node-transit-switch-port-ifaddr` as the port's address
+
+This setup supports only a single VTEP per node.
+
+To support multiple Transit Switch ports per node -- each associated with a specific encap IP, two new annotations
+must be introduced:
+- `k8s.ovn.org/node-transit-switch-port-tunnel-ids`:  
+  Stores the tunnel IDs allocated for each Transit Switch port.
+- `k8s.ovn.org/node-transit-switch-port-ifaddrs`:  
+  Stores the IP addresses assigned to each Transit Switch port.
+
+These annotations are added by the ovnkube-cluster-manager when a node joins the cluster. The number of tunnel IDs
+and addresses must match the number of encap IPs configured on the node.
```

Comment:
IIUC, only a subset of VTEPs is configured for every NAD. So transit IPs and tunnel keys don't need to be unique across all VTEPs, but only within the same NAD?
So if you have 8 VTEPs, and you allocate 3 of them for NAD1, then you don't need annotations for all 8 of them right?